exonum/src/blockchain/service.rs

```diff
@@ -148,6 +148,18 @@ pub trait Service: Send + Sync + 'static {
     fn state_hash(&self, snapshot: &Snapshot) -> Vec<Hash>;
 
     /// Tries to create a `Transaction` from the given raw message.
+    ///
+    /// Service should return an error in the following cases (see `MessageError` for more details):
+    /// - Incorrect transaction identifier.
+    /// - Incorrect data layout.
+    ///
+    /// Service _shouldn't_ perform signature check or logical validation of the transaction: these
+    /// operations should be performed in the `Transaction::verify` and `Transaction::execute`
+    /// methods respectively.
```

Comment:
Nitpick: `verify` may perform logical validation of the transaction (provided it does not depend on external factors), so I'd remove "respectively".